sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -87,7 +110,10 @@ def sdc_pandas_dataframe_reduce_columns(df, func_name, params, ser_params):
 
     df_func_name = f'_df_{func_name}_impl'
 
-    func_text, global_vars = _dataframe_reduce_columns_codegen(func_name, all_params, s_par, df.columns)
+    if func_name in funcs_result_series:
```

Comment:
Extremely bad idea. Function behavior depends on some global object. Very opaque and confusing.

If you don't want to duplicate code it is better to add additional parameter to `sdc_pandas_dataframe_reduce_columns` function